spec/03-plugins/26-oauth2/03-access_spec.lua

```diff
@@ -1725,6 +1725,38 @@ describe("Plugin: oauth2 (access)", function()
       })
       assert.res_status(200, res)
     end)
+    it("works when a correct access_token is being sent in a form body (PUT)", function()
+      local token = provision_token()
+
+      local res = assert(proxy_ssl_client:send {
+        method = "PUT",
+        path = "/request",
+        body = {
+          access_token = token.access_token
+        },
+        headers = {
+          ["Host"] = "oauth2.com",
+          ["Content-Type"] = "application/json"
+        }
+      })
+      assert.res_status(200, res)
+    end)
+    it("works when a correct access_token is being sent in a form body (PATCH)", function()
+      local token = provision_token()
+
+      local res = assert(proxy_ssl_client:send {
+        method = "PATCH",
+        path = "/request",
+        body = {
+          access_token = token.access_token
+        },
+        headers = {
+          ["Host"] = "oauth2.com",
+          ["Content-Type"] = "application/json"
+        }
+      })
+      assert.res_status(200, res)
+    end)
```

Comment:
Those tests aren't failing at all when I try them against the current master? Are we sure they are representative of the error encountered in #3055? For them to qualify as regression tests, we need to demonstrate the observed issue in those tests, and ensure that this patch fixes it.

Reply:
I think what we should test is the presence (or not) of the error log observed by #3055? We could read the logs in `servroot/logs/error.log` and parse them to see if this error exists. Unfortunately, we do not have many tests following this pattern, and we tend to ignore those tests rather than trying to write them. I'd be willing to make another exception until we come up with a more "native" solution to do so.